src/Controller/TrickController.php

```diff
@@ -18,4 +18,14 @@ public function getAllTricks(TrickRepository $trickRepository)
             'tricks' => $tricks
         ]);
     }
+
+    #[Route('/trick/{id}', name: 'trick')]
+    public function getTrick(TrickRepository $trickRepository, int $id)
```

Comment:
try directly 
public function getTrick(Trick $trick)

It will use the id in the route to find the right Trick